src/ApprenticeAan/SFA.DAS.ApprenticeAan.Api/Controllers/LocationsController.cs

```diff
@@ -0,0 +1,27 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.ApprenticeAan.Application.Locations.Queries.GetAddresses;
+
+namespace SFA.DAS.ApprenticeAan.Api.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class LocationsController : ControllerBase
+{
+    private readonly IMediator _mediator;
+
+    public LocationsController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> GetAddresses([FromQuery] string query)
+    {
+        var queryResponse = await _mediator.Send(new GetAddressesQuery(query));
+
```

Comment:
I would suggest if (queryResponse==null) return BadRequest() here, and add coverage if you do.

Reply:
It is not expected that the query response will be null, it can be empty list if there is no match.